### ISSUE TITLE
Front matter not being detected

### ISSUE THREAD

**Issue by thompson-tomo** [textlint/textlint#1928]
I recently wrote my own rule to do some validation (https://github.com/open-telemetry/semantic-conventions/pull/2971), however I had difficulty in getting the rule to work. The most painful issue I encountered and is blocking me from having a rule capable of performing fixes is that yaml nodes are not being detected.

My original rule was

```javascript
// rules/heading-title-sync.js
"use strict";

module.exports = function(context) {
  const { Syntax, RuleError, report, fixer, getSource } = context;

  let yamlNode = null;
  let frontMatterTitle = null;
  let h1Node = null;

  return {
    [Syntax.YAML](node) {
      yamlNode = node;
      const raw = getSource(node);
      const match = raw.match(/title:\s*(.+)/);
      if (match) {
        frontMatterTitle = match[1].trim();
      }
    },

    [Syntax.Header](node) {
      if (node.depth === 1 && !h1Node) {
        h1Node = node;
      }
    },

    "Document:exit"(node) {
      const fullText = getSource(node);

      // -----------------------------------
      // CASE 1: No YAML front matter at all
      // -----------------------------------
      if (!yamlNode) {
        const title = h1Node
          ? getSource(h1Node).replace(/^#\s*/, "").trim()
          : "TITLE";

        const yamlBlock = `---\ntitle: ${title}\n---\n\n`;

        report(
          node,
          new RuleError("Missing YAML front matter", {
            fix: fixer.insertTextBeforeRange([0, 0], yamlBlock)
          })
        );
        return;
      }

      // -----------------------------------
      // CASE 2: YAML exists but no title
      // -----------------------------------
      if (!frontMatterTitle) {
        const title = h1Node
          ? getSource(h1Node).replace(/^#\s*/, "").trim()
          : "TITLE";

        const raw = getSource(yamlNode);
        const lines = raw.split("\n");
        lines.splice(1, 0, `title: ${title}`);

        report(
          yamlNode,
          new RuleError("Missing title in YAML front matter", {
            fix: fixer.replaceText(yamlNode, lines.join("\n"))
          })
        );
        return;
      }

      // -----------------------------------
      // CASE 3: YAML title exists but H1 missing
      // -----------------------------------
      if (!h1Node) {
        const insertPos = yamlNode.range[1];
        const h1 = `\n# ${frontMatterTitle}\n\n`;

        report(
          node,
          new RuleError("Missing H1 heading", {
            fix: fixer.insertTextAfterRange([insertPos, insertPos], h1)
          })
        );
        return;
      }

      // -----------------------------------
      // CASE 4: Both exist but differ → update YAML ONLY
      // -----------------------------------
      const h1Text = getSource(h1Node).replace(/^#\s*/, "").trim();

      if (frontMatterTitle !== h1Text) {
        const raw = getSource(yamlNode);
        const updated = raw.replace(
          /title:\s*(.+)/,
          `title: ${h1Text}`
        );

        report(
          yamlNode,
          new RuleError(
            `YAML title (“${frontMatterTitle}”) does not match H1 (“${h1Text}”)`,
            {
              fix: fixer.replaceText(yamlNode, updated)
            }
          )
        );
      }
    }
  };
};
```

However the yaml node was never detected.

the start of my file is:

```
---
title: Android
---

# Android

• Android Attributes
• Deprecated Android Attributes

## Android Attributes
```

But i am getting the following: 
- Syntax: Document
- Syntax: DocumentExit 
- Syntax: Paragraph
- Syntax: ParagraphExit
- Syntax: BlockQuote
- Syntax: BlockQuoteExit
- Syntax: ListItem
- Syntax: ListItemExit
- Syntax: List
- Syntax: ListExit
- Syntax: Header 
- Syntax: HeaderExit
- Syntax: CodeBlock
- Syntax: CodeBlockExit 
- Syntax: HtmlBlock
- Syntax: HtmlBlockExit

I would expect that yaml syntax is identified correctly.

**Comment by azu** [comment]
https://textlint.org/astexplorer/#/snippet/woXCqHBhcnNlcklEwrh0ZXh0bGludDptxINrZG93bi10by1hc3TCqMSFdHTEkGdzwoHEisSMxI7EkMSSxJRyxJbEmMSaxJzEnsSgw4DCqHbEhnNpb27EqMSqxI3Ej8SRxJPElcSXxJnEm8SdxJ90wqYxNS7FjjHCqGZpbGVuYW1lwrBzb3VyY2UudW5kZcWTbmVkwqRjb8WlwrEtxbAKxKXEjmU6IGFhCsWwLQ%7D
front matter is not CommonMark/other language has not front matter.
So, `Syntax` does not have `Yaml`.
You can use `"Yaml"` instead of `Syntax.Yaml`
